kroxylicious/src/main/java/io/kroxylicious/proxy/internal/filter/BrokerAddressFilter.java

```diff
@@ -0,0 +1,98 @@
+import io.kroxylicious.proxy.filter.KrpcFilterContext;
+import io.kroxylicious.proxy.filter.KrpcFilterState;
+import io.kroxylicious.proxy.filter.MetadataResponseFilter;
+
+/**
+ * A filter that rewrites broker addresses in all relevant responses to the corresponding proxy address.
+ */
+public class BrokerAddressFilter implements MetadataResponseFilter, FindCoordinatorResponseFilter, DescribeClusterResponseFilter {
+
+    private static final Logger LOGGER = LogManager.getLogger(BrokerAddressFilter.class);
+
+    public interface AddressMapping {
+        String host(String host, int port);
+
+        int port(String host, int port);
+    }
+
+    private final AddressMapping mapping;
+
+    public BrokerAddressFilter(AddressMapping mapping) {
+        this.mapping = mapping;
+    }
+
+    @Override
+    public KrpcFilterState onMetadataResponse(MetadataResponseData data, KrpcFilterContext context) {
+        mapBrokers(context, data);
+        return KrpcFilterState.FORWARD;
+    }
+
+    @Override
+    public KrpcFilterState onDescribeClusterResponse(DescribeClusterResponseData data, KrpcFilterContext context) {
+        mapBrokers(context, data);
+        return KrpcFilterState.FORWARD;
+    }
+
+    @Override
+    public KrpcFilterState onFindCoordinatorResponse(FindCoordinatorResponseData data, KrpcFilterContext context) {
+        mapCoordinators(context, data);
+        return KrpcFilterState.FORWARD;
+    }
+
+    private void mapBrokers(KrpcFilterContext context, MetadataResponseData data) {
+        for (var broker : data.brokers()) {
+            String host = mapping.host(broker.host(), broker.port());
+            int port = mapping.port(broker.host(), broker.port());
+            LOGGER.trace("{}: Rewriting metadata response {}:{} -> {}:{}", context, broker.host(), broker.port(), host, port);
+            broker.setHost(host);
+            broker.setPort(port);
+        }
+    }
+
+    private void mapBrokers(KrpcFilterContext context, DescribeClusterResponseData data) {
+        for (var broker : data.brokers()) {
+            String host = mapping.host(broker.host(), broker.port());
+            int port = mapping.port(broker.host(), broker.port());
+            LOGGER.trace("{}: Rewriting describe cluster response {}:{} -> {}:{}", context, broker.host(), broker.port(), host, port);
+            broker.setHost(host);
+            broker.setPort(port);
+        }
+    }
+
+    private void mapCoordinators(KrpcFilterContext context, FindCoordinatorResponseData data) {
+        for (Coordinator coordinator : data.coordinators()) {
+            String host = mapping.host(coordinator.host(), coordinator.port());
+            int port = mapping.port(coordinator.host(), coordinator.port());
+            LOGGER.trace("{}: Rewriting find coordinator response {}:{} -> {}:{}", context, coordinator.host(), coordinator.port(), host, port);
+            coordinator.setHost(host);
+            coordinator.setPort(port);
+        }
+    }
```

Comment:
Could we abstract over this better using method references for the getters and setters?

Reply:
Pushed a commit for that. Let me know, I think we can go with it, I like it.